server/src/__tests__/definition.test.ts

```diff
@@ -578,4 +578,86 @@ describe('on definition', () => {
       ])
     )
   })
+
+  it('only provides symbols in declaration expression as definitions', async () => {
+    analyzer.analyze({
+      uri: DUMMY_URI,
+      document: FIXTURE_DOCUMENT.HOVER
+    })
+
+    const shouldWork = onDefinitionHandler({
+      textDocument: {
+        uri: DUMMY_URI
+      },
+      position: {
+        line: 1,
+        character: 1
+      }
+    })
+
+    expect(shouldWork).toEqual(
+      expect.arrayContaining([
+        {
+          uri: DUMMY_URI,
+          range: {
+            start: {
+              line: 1,
+              character: 0
+            },
+            end: {
+              line: 1,
+              character: 11
+            }
+          }
+        }
+      ])
+    )
+
+    // Symbols with the same identifier but with different kind or the ones in ${} should not be definitions
```

Comment:
When I run this branch, I still get references as definitions if I click "go to definition" on the definition (not if I do it on the references). Only for single definitions:
```bitbake
VAR = "123"
VAR2 = "${VAR}
```
Ctrl+Click on VAR
![image](https://github.com/yoctoproject/vscode-bitbake/assets/19970889/a4068af0-7df4-467b-8ac7-3b506637f5aa)

Reply:
There is a property that seems to control this behaviour: `ReferenceParam.ReferenceContext.includeDeclaration` (I already am logging it in the handler), but I haven't figured out how to control it after some research. It is automatically passed to the handler function but there doesn't seem to be a setting for it. 

I will note it down and open another PR for this. The current workaround is just filtering out the symbol that is used to trigger the references/definitions.

Reply:
It's in no way critical. If there's no fix it's not an issue.